### ISSUE TITLE
Plugin search path is inconsistent with plugin install path

### ISSUE THREAD

**Issue by mgorny** [midori-browser/core#165]
Original report: https://bugs.gentoo.org/670354

Apparently midori is searching for plugins in `lib` directory while they are actually installed to `CMAKE_INSTALL_LIBDIR` which is frequently `lib64` on amd64 systems. The code probably needs to be adjusted (substituted?) to account for different values of `CMAKE_INSTALL_LIBDIR`, or alternatively plugins need to be forced to `lib/...`.

**Comment by kalikiana** [comment]
The code path in question was meant to avoid hard-coded paths to retain relocatability ie. copying the binaries into a different prefix or putting them into a portable package (snap, Flatpak).
- If the correct solution is to expose `Config.LIBDIR` via `vapi/config.vapi` this needs to be compatible with relocation - maybe we need both the hard-coded path and a fallback to the current one?
- A `lib` dir that's not changing would be easiest, but maybe go against packaging policy?

**Comment by mgorny** [comment]
I'm sorry but I know nothing about snap or flatpak (and despise both), and certainly not how to account for them.

And yes, technically speaking the most correct solution would be to write all the paths you need at build time (using `${CMAKE_FULL_*DIR}` variables) and use them from the installed package. I suppose executable-relative paths might work (besides being completely non-portable) if you do some complex logic that figures the relative path between `CMAKE_INSTALL_BINDIR` and `CMAKE_INSTALL_LIBDIR`. Of course, then there's the whole deal of potential symbolic links that could completely break that logic…

In other words, I suppose you'd need to have more conditional logic for this. One set of paths in build dir, another written at build time for regular installs and possibly a third one if neither of the two work for flatpak/snap.

That said, you may want to make the plugin path configurable separately from library path. Some people would argue LIBEXECDIR is more correct than LIBDIR for them.

**Comment by kalikiana** [comment]
After a bit of testing it seems that passing the folder from CMake down to `Config.PLUGINDIR` is good enough. And the value can be overridden via `-DPLUGINDIR=` if need be. See #168
